nova/gce/driver.py

```diff
@@ -0,0 +1,803 @@
+                                power_on=True):
+        pass
+
+    def post_live_migration_at_destination(self,
+                                           context,
+                                           instance,
+                                           network_info,
+                                           block_migration=False,
+                                           block_device_info=None):
+        pass
+
+    def power_off(self, instance, timeout=0, retry_interval=0):
+        """
+        Power off the specified instance.
+        :param instance: nova.objects.instance.Instance
+        :param timeout: time to wait for GuestOS to shutdown
+        :param retry_interval: How often to signal guest while
+                               waiting for it to shutdown
+        """
+        compute, project, zone = self.gce_svc, self.gce_project, self.gce_zone
+        gce_id = self._get_gce_id_from_instance(instance)
+        LOG.info('Stopping instance')
+        operation = gceutils.stop_instance(compute, project, zone, gce_id)
+        gceutils.wait_for_operation(compute, project, zone, operation)
+        LOG.info("Power off Complete.")
+
+    def power_on(self, context, instance, network_info, block_device_info):
+        """Power on the specified instance."""
+        compute, project, zone = self.gce_svc, self.gce_project, self.gce_zone
+        gce_id = self._get_gce_id_from_instance(instance)
+        LOG.info('Starting instance')
+        operation = gceutils.start_instance(compute, project, zone, gce_id)
+        gceutils.wait_for_operation(compute, project, zone, operation)
+        LOG.info("Power on Complete.")
+
+    def soft_delete(self, instance):
+        """Deleting the specified instance"""
+        self.destroy(instance)
+
+    def restore(self, instance):
+        pass
+
+    def pause(self, instance):
+        """
+        GCE doesn't support pause and cannot save system state and hence
+        we've implemented the closest functionality which is to poweroff the
+        instance.
+        :param instance: nova.objects.instance.Instance
+        """
+        self.power_off(instance)
+
+    def unpause(self, instance):
+        """
+        Since GCE doesn't support pause and cannot save system state, we
+        had implemented the closest functionality which is to poweroff the
+        instance. and powering on such an instance in this method.
+        :param instance: nova.objects.instance.Instance
+        """
+        self.power_on(
+            context=None,
+            instance=instance,
+            network_info=None,
+            block_device_info=None)
+
+    def suspend(self, context, instance):
+        """
+        GCE doesn't support suspend and cannot save system state and hence
+        we've implemented the closest functionality which is to poweroff the
+        instance.
+        :param instance: nova.objects.instance.Instance
+        """
+        self.power_off(instance)
```

Comment:
Can you just add a log here suggesting power_off is being requested instead of suspend for clarity?